📖 Add a security self assessment doc (#876)
open-cluster-management-io/ocm: SELF_ASSESSMENT.md

### Responsible Disclosure Practice

The OCM project accepts vulnerability reports through the email [OCM-security@googlegroups.com](mailto:OCM-security@googlegroups.com), a maintainer will collaborate directly with the reporter through the email or Slack direct message until it is resolved.

### Incident Response

In the event that a vulnerability is reported, the maintainer team will collaborate to determine the validity and criticality of the report. Based on these findings, the fix will be triaged and the maintainer team will work to issue a patch in a timely manner.

Patches will be made to the most recent three minor releases. Information will be disseminated to the community through all appropriate outbound channels as soon as possible based on the circumstance.

## Appendix

- Known Issues Over Time
  - There are currently no known vulnerabilities in any version.
- OpenSSF Best Practices
  - OCM has attained the Open Source Security Foundation(OpenSSF) Best Practices Badge, refer to https://bestpractices.coreinfrastructure.org/projects/5376.
- Case Studies

>   - All apoters can be found at [adopters-list](https://github.com/open-cluster-management-io/ocm/blob/main/ADOPTERS.md).

Reviewer: ```suggestion
  - All adopters can be found at [adopters-list](https://github.com/open-cluster-management-io/ocm/blob/main/ADOPTERS.md).
```

The OCM security policy is maintained in the website [Security page](https://open-cluster-management.io/docs/security/).

### Responsible Disclosure Practice

The OCM project accepts vulnerability reports through the email [OCM-security@googlegroups.com](mailto:OCM-security@googlegroups.com), a maintainer will collaborate directly with the reporter through the email or Slack direct message until it is resolved.

### Incident Response

In the event that a vulnerability is reported, the maintainer team will collaborate to determine the validity and criticality of the report. Based on these findings, the fix will be triaged and the maintainer team will work to issue a patch in a timely manner.

Patches will be made to the most recent three minor releases. Information will be disseminated to the community through all appropriate outbound channels as soon as possible based on the circumstance.

## Appendix

- Known Issues Over Time
  - There are currently no known vulnerabilities in any version.
- OpenSSF Best Practices
  - OCM has attained the Open Source Security Foundation(OpenSSF) Best Practices Badge, refer to https://bestpractices.coreinfrastructure.org/projects/5376.

> - Case Studies

Author: Case Studies are added, @qiujian16 @mikeshng PTAL

## Table of Contents

- [Open Cluster Management Self-Assessment](#open-cluster-management-self-assessment)
  - [Table of Contents](#table-of-contents)
  - [Metadata](#metadata)
  - [Overview](#overview)
    - [Background](#background)
    - [Actors](#actors)
    - [Actions](#actions)
      - [Register a managed cluster](#register-a-managed-cluster)
      - [Detach a managed cluster](#detach-a-managed-cluster)
      - [Workload distribution](#workload-distribution)
    - [Goals](#goals)
    - [Non-Goals](#non-goals)
  - [Self-assessment Use](#self-assessment-use)
  - [Security functions and features](#security-functions-and-features)
  - [Project Compliance](#project-compliance)
  - [Secure Development Practices](#secure-development-practices)
    - [Deployment Pipeline](#deployment-pipeline)
    - [Communication Channels](#communication-channels)
  - [Security Issue Resolution](#security-issue-resolution)
    - [Responsible Disclosure Practice](#responsible-disclosure-practice)
    - [Incident Response](#incident-response)
  - [Appendix](#appendix)

## Metadata

> | | |

Reviewer: ```suggestion
```

Remove this blank row?

Author: The table header needs to be kept, otherwise Markdown cannot render the table.

### Deployment Pipeline

In order to secure the SDLC from development to deployment, the following measures are in place.

All code is maintained on [Github](https://github.com/open-cluster-management-io/ocm).

- Contributions and Changes
  - Code changes are submitted via Pull Requests (PRs) and must be signed and verified.
  - Commits to the main branch directly are not allowed.
- Code Review
  - Changes must be reviewed by at least 1 reviewer.
  - Chagees must be approved by at least 1 maintainers.
- Automated Testing
  - In each PR, the code has to pass through linting verify and various security checks and vulnerability analysis, to find if the code is secure and would not fail basic testing.
  - Tools like Dependency Review, License Compliance have been adopted for security scanning.
  - The project utilizes various unit tests and e2e tests to quantify whether the changes would be safe in basic context, before the reviews done by the project maintainers.
- Dependency Management
  - The project regularly updates its dependencies and check for vulnerabilities and keeps its github updated at all times asynchronously.

### Communication Channels

> Internal communications among OCM maintainers and contributors are handled through the public [Slack channel](https://kubernetes.slack.com/archives/C01GE7YSUUF) and direct messages. Inbound communications are accepted through [GitHub Issues](https://github.com/open-cluster-management-io/ocm/issues) or the public [Slack channel](https://kubernetes.slack.com/archives/C01GE7YSUUF) and direct messages. Outbound messages to users are made primarily via documentation or release notes, and secondarily via the public [Slack channel](https://kubernetes.slack.com/archives/C01GE7YSUUF).

Reviewer: ```suggestion
Internal communications among OCM maintainers and contributors are handled through the public [Slack channel](https://kubernetes.slack.com/channels/open-cluster-mgmt) and direct messages. Inbound communications are accepted through [GitHub Issues](https://github.com/open-cluster-management-io/ocm/issues) or the public [Slack channel](https://kubernetes.slack.com/channels/open-cluster-mgmt) and direct messages. Outbound messages to users are made primarily via documentation or release notes, and secondarily via the public [Slack channel](https://kubernetes.slack.com/channels/open-cluster-mgmt).
```
